### PR TITLE
Gsdx refresh gui options

### PR DESCRIPTION
**Pull request by gregory38**
I removed the safe accurate blending option. Not sure it plays well on windows.

### CODE REVIEW

**Comment by FlatOutPS2** [comment]
> I removed the safe accurate blending option. Not sure it plays well on windows.

What could be the issue?

**Comment by gregory38** [comment]
I just hidded the option but it is still here. But I didn't test anything on windows.

**Comment by FlatOutPS2** [comment]
It leaves a gap in the Windows hack UI, but it seems fine otherwise.

**Comment by gregory38** [comment]
A patch is welcome to fix properly the Windows gui ;)

**Comment by FlatOutPS2** [comment]
```
 plugins/GSdx/GSdx.rc | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/plugins/GSdx/GSdx.rc b/plugins/GSdx/GSdx.rc
index 88e28f3..ad1135b 100644
--- a/plugins/GSdx/GSdx.rc
+++ b/plugins/GSdx/GSdx.rc
@@ -128,8 +128,8 @@ BEGIN
     CONTROL         "Half-pixel Offset",IDC_OFFSETHACK,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,51,70,8
     COMBOBOX        IDC_MSAACB,88,17,58,63,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
     CONTROL         "Wild Arms Offset",IDC_WILDHACK,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,64,70,8
-    CONTROL         "Safe Accurate Blending",IDC_SAFE_FBMASK,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,77,90,8
-    CONTROL         "Alpha Stencil",IDC_ALPHASTENCIL,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,77,57,8
+    CONTROL         "Unscale Point Line", IDC_UNSCALE_POINT_LINE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,77,90,8
+    CONTROL         "Alpha Stencil",IDC_ALPHASTENCIL,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,88,77,57,8
     CONTROL         "Align Sprite",IDC_ALIGN_SPRITE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,88,51,58,8
     CONTROL         "Disable Depth Emulation",IDC_TC_DEPTH,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,90,92,8
     RTEXT           "TC Offset X:",IDC_STATIC,40,179,44,8
@@ -142,8 +142,7 @@ BEGIN
     RTEXT           "Sprite:",IDC_SPRITEHACK_TEXT,62,160,22,8
     COMBOBOX        IDC_SPRITEHACK,88,158,58,63,CBS_DROPDOWNLIST | WS_VSCROLL | WS_TABSTOP
     CONTROL         "Fast Texture Invalidation",IDC_FAST_TC_INV,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,102,92,8
-    CONTROL         "Auto Flush", IDC_AUTO_FLUSH,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,88,128,45,8
-    CONTROL         "Unscale Point Line", IDC_UNSCALE_POINT_LINE,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,128,70,8
+    CONTROL         "Auto Flush", IDC_AUTO_FLUSH,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,14,128,70,8
     RTEXT           "TC Offset Y:",IDC_STATIC,40,197,44,8
 END
```

**Comment by jcdenton2k** [comment]
Having tested the 'safe accurate blending' on Windows it does dramatically improve certain aspects (with little to no performance loss) on my setup for FF12 IZJS.  I've noticed sandstorms and other 'fog' style effects are reproduced with greater accuracy than before.

It seems a shame to remove this option when considering it is already hidden away in the 'Hacks' submenu.  Looks like I'll stay on my current version until this gets reintroduced.

**Comment by gregory38** [comment]
Better how? It ought to be equivalent. What is your gpu and driver?

**Comment by jcdenton2k** [comment]
CPU: AMD Phenom II X4 840
GPU: Nvidia GTX 560 Ti
Latest Nvidia drivers.

I'll grab some screenshots with/without if you like but I'd prefer that safe accurate blending not be removed unless it is useless (not working) or causes issues/bugs in other things.

I posted here to say that it does work (for me) in slightly improving some fog-style effects like sandstorms and certain spell effects in FF12 IZJS.  I'm sure it'll also work in FF12 USA.

If more evidence is required, I might pull out my Silent Hill 2 PS2 disc and rip that to an ISO to test out the fog effects in that game with/without this hack.

**Comment by jcdenton2k** [comment]
Ok, after reading what 'safe accurate blending' actually does, I understand why it was removed.  Disregard my previous posts and apologies for bugging you about it.

http://pcsx2.net/274-jan-feb-2016-progress-report.html

Turns out that 'Safe' Accurate Blending is just a slower version of what is being done by default in newer builds anyways.  It was only left in as a testing option and it seems testing has been completed so that option is no longer necessary.

Feel free to remove it.  I'll do more research before poking my head in for future posts.

**Comment by FlatOutPS2** [comment]
Well, if the Safe option really is the only way to get some of the FFXII spells and sandstorms to display correctly, it shouldn't get removed just yet.

File: plugins/GSdx/GSSetting.cpp [hunk 1]
```diff
@@ -102,9 +102,6 @@ const char* dialog_message(int ID, bool* updateText) {
 				"High:\nExtend it to destination alpha blending and color wrapping. (help shadow and fog effect). A good CPU is required.\n\n"
 				"Full:\nExcept few cases, the blending unit will be fully emulated by the shader. It is ultra slow! It is intended for debug.\n\n"
 				"Ultra:\nThe blending unit will be completely emulated by the shader. It is ultra slow! It is intended for debug.";
-		case IDC_SAFE_FBMASK:
```
Comment (ssakash) [inline plugins/GSdx/GSSetting.cpp]: Maybe it's enum declaration could also be removed from ``GSSetting.h`` ? (Linux only)

Comment (ssakash) [inline plugins/GSdx/GSSetting.cpp]: Also on windows side - If you're planning on removing safe accurate blending then make sure to also remove the ``IDC_SAFE_FBMASK`` define from [Resource file](https://github.com/PCSX2/pcsx2/blob/master/plugins/GSdx/resource.h#L104).

**Comment by gregory38** [comment]
@jcdenton2k to close the topic. You confirm me that FFXII is fine with accurate blending enabled. And the safe option can be removed.

**Comment by jcdenton2k** [comment]
I'm going to test this out right now and add in my results.  I'll be using OpenGL renderer with PCSX2 v1.5.0-dev-1319-gb00ae97 compiled Sept-20-2016.  I'll edit this post once my results are finished.

**Comment by FlatOutPS2** [comment]
I checked up on a bug I noticed while playing FFXII a few months ago, but it's solved by Preload Frame Data. So it seems there is no known issue that's solved by using Safe Accurate Blending.

**Comment by gregory38** [comment]
```
With the way emulation can be unpredictable, it may be useful to maintain the 'safe accurate blending' as an option and just rename it 'slow accurate blending' to reflect the functionality better.
```
Well it's still working as expected. My main jobs is hardware stuff. And I can't find a way that could break the HW.

**Comment by gregory38** [comment]
@ssakash please review your review ;)

**Comment by gregory38** [comment]
Maybe you will do it when you will add a mipmap option :)

**Comment by ssakash** [comment]
> Maybe you will do it when you will add a mipmap option :)

I'm not the only windows contributor here :P I'll let our trusty @FlatOutPS2 handle it. I think I had enough fun with Windows specific GUI changes.

**Pull request merged**